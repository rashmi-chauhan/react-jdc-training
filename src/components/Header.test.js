import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from './Header';

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it('should renders welcome', () => {
    const component = renderer.create(<Header />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render properly', () => {
    const header = shallow(<Header />);
    //expect(header.find({ prop: 'title' })).to.have.length(1);
    //expect(header.find(<h2/>)).to.have.length(1);
    expect(header.find('div h2').text()).toEqual('Welcome to React JDC Training');
  });

  it('should handle a click event', () => {
      const wrapper = mount(<a>Click Me</a>, {
        context: {color: 'blue'}
      })
      expect(mountToJson(wrapper)).toMatchSnapshot()
  });
});