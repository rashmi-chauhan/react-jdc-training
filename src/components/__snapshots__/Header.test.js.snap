// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders welcome 1`] = `
<div
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <a
    href="/"
    id="logo"
    onClick={undefined}
  >
    <h2 />
  </a>
</div>
`;
